lib/puppet/provider/user/useradd.rb

```diff
@@ -203,7 +203,10 @@ def deletecmd
     else
       cmd = [command(:delete)]
     end
-    cmd += @resource.managehome? ? ['-r'] : []
+    # Solaris `useradd -r` will fail if the homedir does not exist.
```

Comment:
Oh, this comment in particular is wrong, it's invoking `userdel -r`.

Reply:
Thanks, @MikaelSmith; fixed.